### DISCUSSION TITLE
CapaSQLite.isJsonValidcitor SQLite.isJsonValid() issue

### DISCUSSION THREAD

**Discussion by storynana** [capacitor-community/sqlite#369] [Q&A] [answered]
Hi, there;
     I am using this plugin version 4.6.0 in one of my ionic project, the CapaSQLite.isJsonValidcitor SQLite.isJsonValid() gives Error: IsJsonValid: Stringify Json Object not Valid. 

   My code looks like this:
Test 1:
```
  of(dbJson).subscribe(
        async (jsonExport: JsonSQLite) => {
        const jsonstring = JSON.stringify(jsonExport);
       const isValid = await CapacitorSQLite.isJsonValid({ jsonstring: jsonstring});
       ----other code-----
 )
```
where the "dbJson" is imported from a valid Json data schema file. 
The CapacitorSQLite.isJsonValid gives error "Error: IsJsonValid: Stringify Json Object not Valid"

 I tried another Json schema file which works well in my another project with plugin version 4.1.0, but version 4.6.0 gives the error.

Then I tried using the sample data  which is jsonSQLite  object in "https://github.com/capacitor-community/sqlite/blob/master/docs/ImportExportJson.md", then tested in the code below  which works ok:

Test 2:
```
 const jsonstring = JSON.stringify(dataToImport);.
 const isValid = await CapacitorSQLite.isJsonValid({ jsonstring: jsonstring});

```
 where the "dataToImport" is the sample data (jsonSQLite  object) provided in the above link.

Not finished yet. 

Then I replaced the Json file with the string generated from JSON.stringify(dataToImport) and then retried from my code  in "Test 1", 
 Interesting thing is that CapacitorSQLite.isJsonValid still gives error "Error: IsJsonValid: Stringify Json Object not Valid".

Can some please help this?

Cheers,
Qiuwei

**Comment by jepiqueau** [comment]
hi, can you show a bit more code I may assume that of(dbJson) is an observable so can you give how it is define?

**Reply by storynana** [reply]
HI, jepiqueau;

sure. "dbJson" is an imported json data schema file.

```
import * as dbJson  from '../../assets/database/testData-schema.db.json'

----other code-----

private initDB() {
	of(dbJson).subscribe(
		  async (jsonExport: JsonSQLite) => {
			const jsonstring = JSON.stringify(jsonExport);
			console.log(jsonstring);
			const isValid = await CapacitorSQLite.isJsonValid({ jsonstring: jsonstring });

			if (isValid.result) {
			  console.log("--------before-------");
			  await CapacitorSQLite.importFromJson({ jsonstring });
			  console.log("--------after-------");
			}
      }
}
```
where testData-schema.db.json (converted from JsonSQLite object nin https://github.com/capacitor-community/sqlite/blob/master/docs/ImportExportJson.md")
[testData-schema.db.txt](https://github.com/capacitor-community/sqlite/files/10406064/testData-schema.db.txt)

By the way, I don't have to use "of(dbJson)" to make it as observable, I tried but it gives same error.

Please change the file extension from "txt" to "json", github doesn't allow upload json file directly.

Cheers,
Qiuwei

**Comment by jepiqueau** [comment]
@storynana i am away from any computer till Sunday i will have a look after this. In the mean time you can have a look to angular-sqlite-starter-app on https://github.com/jepiqueau there is some example of import export json under the tests folder

**Comment by storynana** [comment]
Thanks Jean.

No hush. Will check it out.

On Fri, Jan 13, 2023 at 8:47 PM QUEAU Jean Pierre ***@***.***>
wrote:

> @storynana <https://github.com/storynana> i am away from any computer
> till Sunday i will have a look after this. In the mean time you can have a
> look to angular-sqlite-starter-app on https://github.com/jepiqueau there
> is some example of import export json under the tests folder
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/capacitor-community/sqlite/discussions/369#discussioncomment-4675365>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACQQ76TX4NTC5EOVS2OJZ7LWSECCFANCNFSM6AAAAAATYTHCRU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Cheers,
qiuwei

**Comment by jepiqueau** [comment, marked as answer]
@storynana  Thank for waiting
i did that
 - copy the [testData-schema.db.txt](https://github.com/capacitor-community/sqlite/files/10406064/testData-schema.db.txt) in the assets/databases folder of one of my app and rename it as .json file
 - add in the `tsconfig.json` inside the `compilerOptions`

```
    "resolveJsonModule": true,
    "allowSyntheticDefaultImports": true,
```

  - edit my home page with
```
   ...
import { of } from 'rxjs';
import dbJson from 'src/assets/databases/testData-schema.db.json';
import { JsonSQLite, CapacitorSQLite } from '@capacitor-community/sqlite';
  ...

ngOnInit() {
  ...
    of(dbJson).subscribe(
      async (jsonExport: JsonSQLite) => {
      const jsonstring = JSON.stringify(jsonExport);
      console.log(jsonstring);
      const isValid = await CapacitorSQLite.isJsonValid({ jsonstring: jsonstring });
      console.log(`-- after is JsonValid isValid.result: ${isValid.result}`);
      if (isValid.result) {
        console.log("--------before-------");
        await CapacitorSQLite.importFromJson({ jsonstring });
        console.log("--------after-------");
      }
    });
  ...
  }
```

and i got in the console
```
home.page.ts:35 {"database":"db-from-json","version":1,"encrypted":false,"mode":"full","tables":[{"name":"users","schema":[{"column":"id","value":"INTEGER PRIMARY KEY NOT NULL"},{"column":"email","value":"TEXT UNIQUE NOT NULL"},{"column":"name","value":"TEXT"},{"column":"age","value":"INTEGER"},{"column":"sql_deleted","value":"BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1))"},{"column":"last_modified","value":"INTEGER DEFAULT (strftime('%s', 'now'))"}],"indexes":[{"name":"index_user_on_name","value":"name"},{"name":"index_user_on_last_modified","value":"last_modified DESC"},{"name":"index_user_on_email_name","value":"email ASC, name","mode":"UNIQUE"}],"values":[[1,"Whiteley.com","Whiteley",30,0,1582536810],[2,"Jones.com","Jones",44,0,1582812800],[3,"Simpson@example.com","Simpson",69,0,1583570630],[4,"Brown@example.com","Brown",15,0,1590383895]]},{"name":"messages","schema":[{"column":"id","value":"INTEGER PRIMARY KEY NOT NULL"},{"column":"userid","value":"INTEGER"},{"column":"title","value":"TEXT NOT NULL"},{"column":"body","value":"TEXT NOT NULL"},{"column":"sql_deleted","value":"BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1))"},{"column":"last_modified","value":"INTEGER DEFAULT (strftime('%s', 'now'))"},{"foreignkey":"userid","value":"REFERENCES users(id) ON DELETE CASCADE"}],"indexes":[{"name":"index_messages_on_title","value":"title"},{"name":"index_messages_on_last_modified","value":"last_modified DESC"}],"values":[[1,1,"test post 1","content test post 1",0,1587310030],[2,2,"test post 2","content test post 2",0,1590388125],[3,1,"test post 3","content test post 3",0,1590383895]]},{"name":"images","schema":[{"column":"id","value":"INTEGER PRIMARY KEY NOT NULL"},{"column":"name","value":"TEXT UNIQUE NOT NULL"},{"column":"type","value":"TEXT NOT NULL"},{"column":"size","value":"INTEGER"},{"column":"img","value":"BLOB"},{"column":"sql_deleted","value":"BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1))"},{"column":"last_modified","value":"INTEGER DEFAULT (strftime('%s', 'now'))"}],"indexes":[{"name":"index_images_on_last_modified","value":"last_modified DESC"}],"values":[[1,"meowth","png","NULL","data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAAAkCAYAAAD7PHgWAAAEcElEQVRYR8WYP2hTQRzHfx10aQchi0JcLGpBSBcrlTrpIjoFiy6FDipOHVz8Q0HrUGxdg1N1KBRBackiVoQ6FMVIuzQgpEpdjOiSLUXQIfK9976X37t3l6RNxVuS3Hvv7nPf3+/3vcvraTQaDdlFK4z3yMT8rh7d0Ww97QAzfX12wFq9br4buOk7UpicaQm5F4toCajh9LKnLm23Bex0Ee3k7ArwS/mVvH5elqEzzWmGr0dhDwGGFs3ouMAdA7491y+Dhw5KZuG9UEEA1r6XZfhUPOxgQ0pzPQJIDTi11NtOKOkKkHCcpfDrjQlxaXnGdFE1fAcg2to7sWmgAfVYWCzbPwO06imNHt0Tyd/IyfDlrYRy7kI3fvyUsyvRPbsCxIPIGQ6MAdFWD5RbKnjxZhTSWn0+AqyuS2agEPWNjZhPjrUngBgQkABDQ3hNOJdnmvkXa5UZ6W2CxXBaRoBiLLR2cLgnUSRIbOSLlptVx8LQk7k5iHutah44Pks12+VfApBVh04YsAbV1yR7sslYXU+oSPUK46NWZWPmseJdATLfTJ5UJsxYBNXqoc+EeX7RgpbmRmX1pcjsSq95VkP5AM1czMl63ViS27iNen2QYSUoH+bWVq1WpTh5OAFp1ekbtz7JRVJBPH/+Sk6O5i4YQCxc57Sbq0i1loA2R6hKfDho7rFLqZWzYvXiqCKgSi/6LSC+o7l2ZCIWz5UChHqfH2alvPVVRp/sT4Q7P/1NstmssZ6okNKAyD803+5BICjohjm90qgnAajhcNEHiP7BgQHZqFQkK49FF40uDtyHrZAKEQ6/NWDIoAkcBAQcmpuHoZWG+l1IwlHBjgGp3rP1zchi4kpG3vi+7wQUkMgz5p8tKIwdnzHbhtiatALTRcLvtBnmmc/ANQCuo3JxLGMF6+tmHFUULqgJsUl6Bwy/jXr1elQUWlGnj37JyfQksBhWL/tpM/itK9kHanOQ3rd47bcZxxSIkl97ow67u2Lfouh/+l6EnIvXuU5/TNkMAAjnA7RhUf9RQkWkTRhh9TUCuuO6kUooCMBc/xHzzLG71ZYJjAUhPD6TDUERxoXTC7CRiqOXAIRBZ/J5e3/oXxvhdE6FqpA2g+sslFaA3iLRMmvfYz6l8ixWD/3adF0bwXUNiN87gcP9qfOg72jkepVWkIC6ELQZu5BdAWIwbSl6F9AWQEAXRB8GtOpaxa4BCan3Tp3cemJ3G9R+R/g9DbGenDtLCJQVHIL0AeqKb7fFkaWjdzMIrz4+afdvpWKoslks+Lx9YltufQy/hPICUj1OQAOHR9KGeABwAfk6xOeFOmdrxaI5c6Ktffgjs5/4VzV6QRVUkKcafRMHQh8hQ9udPrm4ChJQw7n3EJYp4D0PPl3YlKtjx+0K3UEAiZ3G9T3fATWRd5UJ8cEBCm3o9D47Fc8CKUCEEw/om/kUD7H4zY2e+Vh8UJb8/fTrDt+BA8/rfZ/j63m9gLSYUHL7Ks99ndZpdYZew3Fub4hbVd3/uvYXfqiMwjPten8AAAAASUVORK5CYII=",0,1590388825],[2,"feather","png","NULL","data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAMAAABEpIrGAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAU1QTFRFNjtAQEVK////bG9zSk9T/v7+/f39/f3+9vf3O0BETlJWNzxB/Pz8d3t+TFFVzM3O1NXX7u/vUldbRElNs7W3v8HCmZyeRkpPW19j8vLy7u7vvsDC9PT1cHR3Oj9Eo6WnxsjJR0tQOD1Bj5KVgYSHTVFWtri50dLUtLa4YmZqOT5D8vPzRUpOkZOWc3Z64uPjr7Gzuru95+jpX2NnaGxwPkNHp6mrioyPlZeadXh8Q0hNPEBFyszNh4qNc3d6eHx/OD1Cw8XGXGBkfoGEra+xxcbIgoaJu72/m52ggoWIZ2tu8/P0wcLE+vr7kZSXgIOGP0NIvr/BvL6/QUZKP0RJkpWYpKaoqKqtVVldmJqdl5qcZWhstbe5bHB0bnJ1UVVZwsTF5ubnT1RYcHN3oaSm3N3e3NzdQkdLnJ+h9fX1TlNX+Pj47/DwwsPFVFhcEpC44wAAAShJREFUeNq8k0VvxDAQhZOXDS52mRnKzLRlZmZm+v/HxmnUOlFaSz3su4xm/BkGzLn4P+XimOJZyw0FKufelfbfAe89dMmBBdUZ8G1eCJMba69Al+AABOOm/7j0DDGXtQP9bXjYN2tWGQfyA1Yg1kSu95x9GKHiIOBXLcAwUD1JJSBVfUbwGGi2AIvoneK4bCblSS8b0RwwRAPbCHx52kH60K1b9zQUjQKiULbMDbulEjGha/RQQFDE0/ezW8kR3C3kOJXmFcSyrcQR7FDAi55nuGABZkT5hqpk3xughDN7FOHHHd0LLU9qtV7r7uhsuRwt6pEJJFVLN4V5CT+SErpXt81DbHautkpBeHeaqNDRqUA0Uo5GkgXGyI3xDZ/q/wJMsb7/pwADAGqZHDyWkHd1AAAAAElFTkSuQmCC",0,1590389895]]}]}
home.page.ts:37 -- after is JsonValid isValid.result: true
home.page.ts:39 --------before-------
home.page.ts:41 --------after-------
``` 

meaning that it works fine.
i also checked that the database   `db-from-jsonSQLite.db` was created

**Reply by whitnessme** [reply]
**EDIT: I think I will make a new Q/A; I did more testing on iOS vs Android and getting a different problem on iOS since its says the json IS valid. #415 

Hello! I am having a similar issue in that I can't get my json to be valid. [Here is the repo.](https://github.com/whitnessme/chao-capacitor-app)
I did the steps above with the `tsconfig.json` and making the file an actual .json file but I am still getting the `Stringify Json Object not Valid` from my `importFromJson` and the `isJsonValid` result gives me `false`. 
I didn't want to start a new Q/A since this is so similar, but let me know if I should post a new discussion instead of just replying here.
Here is the relevant code in my component--I am using NextJS with Typescript and Capacitor:

```ts
// src/pages/home/index.tsx
import * as chaoData from '../../seed-data.json';
const mySQLite = new SQLiteConnection(CapacitorSQLite);

// ...
    const loadJSON = async () => {
        let data = JSON.stringify(chaoData)
        const isValid = await CapacitorSQLite.isJsonValid({ jsonstring: data });
        console.log("ISVALID-----------", isValid.result)
        return await mySQLite.importFromJson(data)
    }
```

Here is my JSON data: 
```json
// src/seed-data.json
{
    "database": "chaodb",
    "version": 1,
    "encrypted": false,
    "mode": "full",
    "tables": [
        {
            "name": "chaos",
            "schema": [
                { "column": "id", "value": "INTEGER PRIMARY KEY NOT NULL" },
                { "column": "name", "value": "TEXT NOT NULL" },
                { "column": "hatch_day", "value": "TEXT NOT NULL" },
                { "column": "color", "value": "TEXT NOT NULL" },
                { "column": "alignment", "value": "TEXT NOT NULL" },
                { "column": "evolution", "value": "TEXT NOT NULL" },
                { "column": "life_cycle", "value": "TEXT NOT NULL" },
                { "column": "personality", "value": "TEXT NOT NULL" },
                {
                    "column": "last_modified",
                    "value": "INTEGER DEFAULT (strftime('%s', 'now'))"
                }
            ],
            "values": [
                [1, "Noodles", "2023-04-02", "Two-toned Brown", "Neutral", "None", "Child", "Big eater", 1604396241],
                [2, "Jewel", "2023-01-03", "Monotone Pink", "Hero", "Swim", "Adult", "Chatty", 1604396241],
                [3, "Bubba", "2023-02-01", "Two-toned Black", "Dark", "Power", "Adult", "Curious", 1604396241],
                [4, "Sparky", "2023-05-03", "Monotone Sky-blue", "Neutral", "Fly", "Adult", "Naughty", 1604396241],
                [5, "Sparky", "2023-05-03", "Monotone Sky-blue", "Neutral", "Fly", "Adult", "Naughty", 1604396241]
            ]
        }
    ]
}
``` 

Here is my tsconfig.json file:
```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "allowSyntheticDefaultImports": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}
```

Thank you! Let me know if you need any other information.

**Comment by storynana** [comment]
Hi,  Jean Pierre;
      Thanks Jean for this response.

      Yes, I tried it again with the same config in the tsconfig.json and
it gives the same error.

      Can you please check your @capacitor-community plugin version number,
my one is sqlite 4.6.0
;
[image: image.png]

My other project runs version 4.1.0 which works fine.
[image: image.png]

Cheers,
Qiuwei

On Tue, Jan 17, 2023 at 7:40 AM QUEAU Jean Pierre ***@***.***>
wrote:

> @storynana <https://github.com/storynana> Thank for waiting
> i did that
>
>    - copy the testData-schema.db.txt
>    <https://github.com/capacitor-community/sqlite/files/10406064/testData-schema.db.txt>
>    in the assets/databases folder of one of my app and rename it as .json file
>    - add in the tsconfig.json inside the compilerOptions
>
>     "resolveJsonModule": true,
>     "allowSyntheticDefaultImports": true,
>
>
>    - edit my home page with
>
>    ...
> import { of } from 'rxjs';
> import dbJson from 'src/assets/databases/testData-schema.db.json';
> import { JsonSQLite, CapacitorSQLite } from ***@***.***/sqlite';
>   ...
>
> ngOnInit() {
>   ...
>     of(dbJson).subscribe(
>       async (jsonExport: JsonSQLite) => {
>       const jsonstring = JSON.stringify(jsonExport);
>       console.log(jsonstring);
>       const isValid = await CapacitorSQLite.isJsonValid({ jsonstring: jsonstring });
>       console.log(`-- after is JsonValid isValid.result: ${isValid.result}`);
>       if (isValid.result) {
>         console.log("--------before-------");
>         await CapacitorSQLite.importFromJson({ jsonstring });
>         console.log("--------after-------");
>       }
>     });
>   ...
>   }
>
> and i got in the console
>
> home.page.ts:35 {"database":"db-from-json","version":1,"encrypted":false,"mode":"full","tables":[{"name":"users","schema":[{"column":"id","value":"INTEGER PRIMARY KEY NOT NULL"},{"column":"email","value":"TEXT UNIQUE NOT NULL"},{"column":"name","value":"TEXT"},{"column":"age","value":"INTEGER"},{"column":"sql_deleted","value":"BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1))"},{"column":"last_modified","value":"INTEGER DEFAULT (strftime('%s', 'now'))"}],"indexes":[{"name":"index_user_on_name","value":"name"},{"name":"index_user_on_last_modified","value":"last_modified DESC"},{"name":"index_user_on_email_name","value":"email ASC, ***@***.******@***.***","Brown",15,0,1590383895]]},{"name":"messages","schema":[{"column":"id","value":"INTEGER PRIMARY KEY NOT NULL"},{"column":"userid","value":"INTEGER"},{"column":"title","value":"TEXT NOT NULL"},{"column":"body","value":"TEXT NOT NULL"},{"column":"sql_deleted","value":"BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1))"},{"column":"last_modified","value":"INTEGER DEFAULT (strftime('%s', 'now'))"},{"foreignkey":"userid","value":"REFERENCES users(id) ON DELETE CASCADE"}],"indexes":[{"name":"index_messages_on_title","value":"title"},{"name":"index_messages_on_last_modified","value":"last_modified DESC"}],"values":[[1,1,"test post 1","content test post 1",0,1587310030],[2,2,"test post 2","content test post 2",0,1590388125],[3,1,"test post 3","content test post 3",0,1590383895]]},{"name":"images","schema":[{"column":"id","value":"INTEGER PRIMARY KEY NOT NULL"},{"column":"name","value":"TEXT UNIQUE NOT NULL"},{"column":"type","value":"TEXT NOT NULL"},{"column":"size","value":"INTEGER"},{"column":"img","value":"BLOB"},{"column":"sql_deleted","value":"BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1))"},{"column":"last_modified","value":"INTEGER DEFAULT (strftime('%s', 'now'))"}],"indexes":[{"name":"index_images_on_last_modified","value":"last_modified DESC"}],"values":[[1,"meowth","png","NULL","data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAAAkCAYAAAD7PHgWAAAEcElEQVRYR8WYP2hTQRzHfx10aQchi0JcLGpBSBcrlTrpIjoFiy6FDipOHVz8Q0HrUGxdg1N1KBRBackiVoQ6FMVIuzQgpEpdjOiSLUXQIfK9976X37t3l6RNxVuS3Hvv7nPf3+/3vcvraTQaDdlFK4z3yMT8rh7d0Ww97QAzfX12wFq9br4buOk7UpicaQm5F4toCajh9LKnLm23Bex0Ee3k7ArwS/mVvH5elqEzzWmGr0dhDwGGFs3ouMAdA7491y+Dhw5KZuG9UEEA1r6XZfhUPOxgQ0pzPQJIDTi11NtOKOkKkHCcpfDrjQlxaXnGdFE1fAcg2to7sWmgAfVYWCzbPwO06imNHt0Tyd/IyfDlrYRy7kI3fvyUsyvRPbsCxIPIGQ6MAdFWD5RbKnjxZhTSWn0+AqyuS2agEPWNjZhPjrUngBgQkABDQ3hNOJdnmvkXa5UZ6W2CxXBaRoBiLLR2cLgnUSRIbOSLlptVx8LQk7k5iHutah44Pks12+VfApBVh04YsAbV1yR7sslYXU+oSPUK46NWZWPmseJdATLfTJ5UJsxYBNXqoc+EeX7RgpbmRmX1pcjsSq95VkP5AM1czMl63ViS27iNen2QYSUoH+bWVq1WpTh5OAFp1ekbtz7JRVJBPH/+Sk6O5i4YQCxc57Sbq0i1loA2R6hKfDho7rFLqZWzYvXiqCKgSi/6LSC+o7l2ZCIWz5UChHqfH2alvPVVRp/sT4Q7P/1NstmssZ6okNKAyD803+5BICjohjm90qgnAajhcNEHiP7BgQHZqFQkK49FF40uDtyHrZAKEQ6/NWDIoAkcBAQcmpuHoZWG+l1IwlHBjgGp3rP1zchi4kpG3vi+7wQUkMgz5p8tKIwdnzHbhtiatALTRcLvtBnmmc/ANQCuo3JxLGMF6+tmHFUULqgJsUl6Bwy/jXr1elQUWlGnj37JyfQksBhWL/tpM/itK9kHanOQ3rd47bcZxxSIkl97ow67u2Lfouh/+l6EnIvXuU5/TNkMAAjnA7RhUf9RQkWkTRhh9TUCuuO6kUooCMBc/xHzzLG71ZYJjAUhPD6TDUERxoXTC7CRiqOXAIRBZ/J5e3/oXxvhdE6FqpA2g+sslFaA3iLRMmvfYz6l8ixWD/3adF0bwXUNiN87gcP9qfOg72jkepVWkIC6ELQZu5BdAWIwbSl6F9AWQEAXRB8GtOpaxa4BCan3Tp3cemJ3G9R+R/g9DbGenDtLCJQVHIL0AeqKb7fFkaWjdzMIrz4+afdvpWKoslks+Lx9YltufQy/hPICUj1OQAOHR9KGeABwAfk6xOeFOmdrxaI5c6Ktffgjs5/4VzV6QRVUkKcafRMHQh8hQ9udPrm4ChJQw7n3EJYp4D0PPl3YlKtjx+0K3UEAiZ3G9T3fATWRd5UJ8cEBCm3o9D47Fc8CKUCEEw/om/kUD7H4zY2e+Vh8UJb8/fTrDt+BA8/rfZ/j63m9gLSYUHL7Ks99ndZpdYZew3Fub4hbVd3/uvYXfqiMwjPten8AAAAASUVORK5CYII=",0,1590388825],[2,"feather","png","NULL","data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAMAAABEpIrGAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAU1QTFRFNjtAQEVK////bG9zSk9T/v7+/f39/f3+9vf3O0BETlJWNzxB/Pz8d3t+TFFVzM3O1NXX7u/vUldbRElNs7W3v8HCmZyeRkpPW19j8vLy7u7vvsDC9PT1cHR3Oj9Eo6WnxsjJR0tQOD1Bj5KVgYSHTVFWtri50dLUtLa4YmZqOT5D8vPzRUpOkZOWc3Z64uPjr7Gzuru95+jpX2NnaGxwPkNHp6mrioyPlZeadXh8Q0hNPEBFyszNh4qNc3d6eHx/OD1Cw8XGXGBkfoGEra+xxcbIgoaJu72/m52ggoWIZ2tu8/P0wcLE+vr7kZSXgIOGP0NIvr/BvL6/QUZKP0RJkpWYpKaoqKqtVVldmJqdl5qcZWhstbe5bHB0bnJ1UVVZwsTF5ubnT1RYcHN3oaSm3N3e3NzdQkdLnJ+h9fX1TlNX+Pj47/DwwsPFVFhcEpC44wAAAShJREFUeNq8k0VvxDAQhZOXDS52mRnKzLRlZmZm+v/HxmnUOlFaSz3su4xm/BkGzLn4P+XimOJZyw0FKufelfbfAe89dMmBBdUZ8G1eCJMba69Al+AABOOm/7j0DDGXtQP9bXjYN2tWGQfyA1Yg1kSu95x9GKHiIOBXLcAwUD1JJSBVfUbwGGi2AIvoneK4bCblSS8b0RwwRAPbCHx52kH60K1b9zQUjQKiULbMDbulEjGha/RQQFDE0/ezW8kR3C3kOJXmFcSyrcQR7FDAi55nuGABZkT5hqpk3xughDN7FOHHHd0LLU9qtV7r7uhsuRwt6pEJJFVLN4V5CT+SErpXt81DbHautkpBeHeaqNDRqUA0Uo5GkgXGyI3xDZ/q/wJMsb7/pwADAGqZHDyWkHd1AAAAAElFTkSuQmCC",0,1590389895]]}]}
> home.page.ts:37 -- after is JsonValid isValid.result: true
> home.page.ts:39 --------before-------
> home.page.ts:41 --------after-------
>
> meaning that it works fine.
> i also checked that the database db-from-jsonSQLite.db was created
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/capacitor-community/sqlite/discussions/369#discussioncomment-4700108>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACQQ76T3J2RD5MQQ76UJUNTWSWI2ZANCNFSM6AAAAAATYTHCRU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Cheers,
qiuwei

**Comment by jepiqueau** [comment]
@storynana did you modify the import too. nothing as changed between 4.6.0 and 4.6.1-2 regarding the json object and the import export json. On which plaform are you running? There is nothing i can do more on this as i cannot reproduce the error. The last thing you can do is to share your app on github so i may have a look

**Comment by storynana** [comment]
Hi, Jean;
     It is Ionic, Angular:
[image: image.png]

Thanks Jean for the effort put on this.
Currently I can still use jsonSQLite to import the data schema.

Thanks again.

Cheers,
Qiuwei

On Tue, Jan 17, 2023 at 8:19 PM QUEAU Jean Pierre ***@***.***>
wrote:

> @storynana <https://github.com/storynana> did you modify the import too.
> nothing as changed between 4.6.0 and 4.6.1-2 regarding the json object and
> the import export json. On which plaform are you running? There is nothing
> i can do more on this as i cannot reproduce the error. The last thing you
> can do is to share your app on github so i may have a look
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/capacitor-community/sqlite/discussions/369#discussioncomment-4704059>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACQQ76UUFZJ7DJTS4EI3OKLWSZBXFANCNFSM6AAAAAATYTHCRU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Cheers,
qiuwei

**Comment by jepiqueau** [comment]
@storynana i was asking for which platform Web, Android, iOS, Electron ?

**Comment by storynana** [comment]
HI, Jean;
     The current project I am working on is currently on the Web, Chrome.

     Another project is Android. The isJsonValid method works for both Web
and Android.

Cheers,
Qiuwei

On Tue, Jan 17, 2023 at 10:41 PM QUEAU Jean Pierre ***@***.***>
wrote:

> @storynana <https://github.com/storynana> i was asking for which platform
> Web, Android, iOS, Electron ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/capacitor-community/sqlite/discussions/369#discussioncomment-4705137>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACQQ76WJPD5ZE76HZA376OTWSZSMDANCNFSM6AAAAAATYTHCRU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Cheers,
qiuwei
